src/plugins/controls/public/types.ts

```diff
@@ -46,6 +47,12 @@ export type ControlEmbeddable<
   renderPrepend?: () => ReactNode | undefined;
 };
 
+export abstract class ClearableControlEmbeddable<
```

Comment:
Nit, do you see use cases where control is not clearable? How about just making `clearSelections` part of ControlEmbeddable interface? I am not convinced that a new interface is needed.

Reply:
Each embeddable extended the generic `Embeddable` class and not `ControlEmbeddable` (which is just a type at the moment and not a class/interface). I can try to refactor things to make each embeddable implement `ControlEmbeddable` instead, cause I think you're right that the current implementation isn't very clear....

Reply:
I do very much like the decoupling here that comes with having the `clear` functionality defined outside of the `ControlEmbeddable` Interface. That said, I'm wondering about the specific choice to use an Abstract class here instead of an interface? IMO an interface accomplishes the same thing with less boilerplate and more type safety. You could do this with:

- An interface `IClearableControl` that defines the clear selections method
- A [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html) that checks for the presence of the clear selections method.
- Run the embeddable through the type guard in the `isCompatible` method of the action rather than using `instanceof`.

that way, theoretically any control can just determine if it's clearable or not by implementing that interface.

Reply:
Changed to this solution in 31e64afd573713fe0a290c7f6b4eb8f520c7ee49 - I think it's cleaner and more inline with how we do other typing :+1: